src/DistributedComputations/distributed_architectures.jl

```diff
@@ -10,38 +10,10 @@ import Oceananigans.Utils: sync_device!, tupleit
 ##### Partitioning
 #####
 
-# Form 3-tuple
-regularize_size(sz::Tuple{<:Int, <:Int}, Rx, Ry) = (sz[1], sz[2], 1)
-
-# Infer whether x- or y-dimension is indicated
-function regularize_size(N::Int, Rx, Ry)
-    if Rx == 1
-        return (1, N, 1)
-    elseif Ry == 1
-        return (N, 1, 1)
-    else
-        throw(ArgumentError("We can't interpret the 1D size $N for 2D partitions!"))
-    end
-end
-
-regularize_size(sz::Array{<:Int, 1}) = regularize_size(sz[1])
-regularize_size(sz::Array{<:Int, 2}) = regularize_size((sz[1], sz[2]))
-regularize_size(sz::Tuple{<:Int})    = regularize_size(sz[1])
-
-# For 3D partitions:
-# regularize_size(sz::Tuple{<:Int, <:Int, <:Int}, Rx, Ry) = sz
-
-struct Partition{S, Rx, Ry, Rz}
-    sizes :: S
-    function Partition{Rx, Ry, Rz}() where {Rx, Ry, Rz}
-        new{Nothing, Rx, Ry, Rz}(nothing)
-    end
-    function Partition(sizes::S) where S
-        Rx = size(sizes, 1)
-        Ry = size(sizes, 2)
-        Rz = size(sizes, 3)
-        return new{S, Rx, Ry, Rz}(sizes)
-    end
+struct Partition{Sx, Sy, Sz}
+    Rx :: Sx
+    Ry :: Sy
+    Rz :: Sz
```

Comment:
```suggestion
    x :: Sx
    y :: Sy
    z :: Sz
```

the "R" used to mean "Ranks". I don't think that's what `Rx` means anymore.